### PR TITLE
Story/9772 fault tolerance certificates/9953 grouping migration

### PR DESCRIPTION
**Pull request by KaiTeuber**
## Description
Added a migration for certified persons

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-9953

### CODE REVIEW

**Review by pascal-brause — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService.swift [hunk 1]
```diff
@@ -382,6 +386,37 @@ class HealthCertificateService {
...
+		}
+
+		// reinsert all health certificates will do the job (it uses the new groupingStandardizedName)
+		let originalInitialHealthCertifiedPersonsReadFromStore = initialHealthCertifiedPersonsReadFromStore
+		initialHealthCertifiedPersonsReadFromStore = false
+		let original = store.healthCertifiedPersons
+		healthCertifiedPersons.removeAll()
+		for person in original {
+			person.healthCertificates.forEach { healthCertificate in
+				Log.debug("Will reregister health certificate")
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService.swift:409]:
```suggestion
				Log.debug("Will re-register health certificate")
```

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService.swift [hunk 2]
```diff
@@ -382,6 +386,37 @@ class HealthCertificateService {
...
+			// after leaving mark migration as done
+			store.healthCertifiedPersonsVersion = kCurrentHealthCertifiedPersonsVersion
+		}
+
+		// reinsert all health certificates will do the job (it uses the new groupingStandardizedName)
+		let originalInitialHealthCertifiedPersonsReadFromStore = initialHealthCertifiedPersonsReadFromStore
+		initialHealthCertifiedPersonsReadFromStore = false
+		let original = store.healthCertifiedPersons
+		healthCertifiedPersons.removeAll()
+		for person in original {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService.swift:407]:
```suggestion
		for person in originalHealthCertifiedPersons {
```

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService.swift [hunk 3]
```diff
@@ -382,6 +386,37 @@ class HealthCertificateService {
...
+			person.healthCertificates.forEach { healthCertificate in
+				Log.debug("Will reregister health certificate")
+				registerHealthCertificate(base45: healthCertificate.base45)
+			}
+		}
+		if original != healthCertifiedPersons {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService.swift:413]:
```suggestion
		if originalHealthCertifiedPersons != healthCertifiedPersons {
```

**Review by naveeddotio — APPROVED** [review]
few suggestions

**Pull request merged**